File: lib/plex.ts
import ky from 'ky';
import { getSettings } from './settings';

export async function scanPlexLibrary() {
    const settings = await getSettings();

    if (!settings?.plexUrl || !settings?.plexToken) {
        console.log('Plex settings not configured. Skipping scan.');
        return;
    }

    try {
        // Ensure URL doesn't end with slash
        const baseUrl = settings.plexUrl.replace(/\/$/, '');

        // Trigger library refresh
        // Endpoint: /library/sections/all/refresh
        await ky.get(`${baseUrl}/library/sections/all/refresh`, {
            searchParams: {
                'X-Plex-Token': settings.plexToken
            },
            timeout: 5000
        });

        console.log('Plex library scan triggered successfully.');
    } catch (error: any) {
        console.error('Failed to trigger Plex scan:', error.message);
    }
}

File: backend/app/services/plex.py
import httpx


class PlexClient:
    def __init__(self, base_url: str, token: str, section_id: str | None = None) -> None:
        self.base_url = base_url.rstrip("/")
        self.token = token
        self.section_id = section_id

    async def refresh_library(self) -> None:
        endpoint = "/library/sections/all/refresh"
        if self.section_id:
            endpoint = f"/library/sections/{self.section_id}/refresh"

        url = f"{self.base_url}{endpoint}"
        params = {"X-Plex-Token": self.token}

        async with httpx.AsyncClient(timeout=20) as client:
            response = await client.get(url, params=params)
            response.raise_for_status()
